### PR TITLE
JDK-8296406: ChainingConstructors jtreg test reduce code duplication

### PR DESCRIPTION
**Pull request by MBaesken**
The java/security/Exceptions/ChainingConstructors jdk jtreg test shows quite a lot of code duplication. The test code can be simplified e.g. by using generics. Additionally the test failure output was improved and a test for the constructors of CertPathBuilderException was added.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8296406](https://bugs.openjdk.org/browse/JDK-8296406): ChainingConstructors jtreg test reduce code duplication

### Reviewers
 * [Sean Mullan](https://openjdk.org/census#mullan) (@seanjmullan - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/11016/head:pull/11016` \
`$ git checkout pull/11016`

Update a local copy of the PR: \
`$ git checkout pull/11016` \
`$ git pull https://git.openjdk.org/jdk pull/11016/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 11016`

View PR using the GUI difftool: \
`$ git pr show -t 11016`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/11016.diff">https://git.openjdk.org/jdk/pull/11016.diff</a>

</details>

### CODE REVIEW

**Comment by mcpowers** [comment]
LGTM

**Comment by seanjmullan** [comment]
Add a `noreg-self` label to the JBS issue since this is a fix to the test itself. Also, I would classify this as an Enhancement, and not a Bug.

**Comment by MBaesken** [comment]
> Add a `noreg-self` label to the JBS issue since this is a fix to the test itself. Also, I would classify this as an Enhancement, and not a Bug.

Thanks, this all makes sense ! I adjusted the JBS issue.

**Review by seanjmullan — CHANGES_REQUESTED** [review]

File: test/jdk/java/security/Exceptions/ChainingConstructors.java [hunk 1]
```diff
@@ -37,174 +37,35 @@ public class ChainingConstructors {
...
-        SSLException ssle =
-                new SSLException(cause);
-        if (!ssle.getCause().equals(cause)) {
-            throw new SecurityException("Test 17 failed");
-        }
-        ssle =new SSLException(MSG, cause);
-        if (!ssle.getMessage().equals(MSG) || !ssle.getCause().equals(cause)) {
-            throw new SecurityException("Test 17 failed");
-        }
-*/
```
[the remark this answers is not in the document]
Comment (MBaesken) [reply test/jdk/java/security/Exceptions/ChainingConstructors.java:208]: Hi Sean,  the summary says that the test is about  "java.security Exceptions" so  javax.net.ssl.SSLException.java from another package might not fit in, maybe that's why it was commented out.
Should I still add it ? If so, the  summary comment might need a little bit adjustment, what about this "Test constructors for exception chaining of  security-related exceptions"  ?

[the remark this answers is not in the document]
Comment (seanjmullan) [reply test/jdk/java/security/Exceptions/ChainingConstructors.java:208]: Good point. I think it is ok to make an exception (no pun intended) for `SSLException` for this test. Changing the summary comment to your suggestion is fine with me.

**Comment by MBaesken** [comment]
I added the second bug id, adjusted the summary comment and added SSLException for this test.

**Review by seanjmullan — APPROVED** [review]
Looks good!

**Comment by MBaesken** [comment]
Mark and Sean, thanks for the reviews !
Do I need another review ?
Best regards, Matthias

**Comment by seanjmullan** [comment]
> Mark and Sean, thanks for the reviews ! Do I need another review ? Best regards, Matthias

No, you can integrate now.

**Pull request closed without merging**